### PR TITLE
cli support for hosted

### PR DESCRIPTION
**Pull request by kushalmalani**
## Description

> Describe the purpose of this pull request.

cli support for hosted

## 🎟 Issue(s)

Related astronomer/astro#10594

## 🧪 Functional Testing

> List the functional testing steps to confirm this feature or fix.

## 📸 Screenshots

> Add screenshots to illustrate the validity of these changes.

**HYBRID**

Screenshot 2023-03-29 at 11 41 05

**HOSTED**

Screenshot 2023-03-29 at 12 43 06

## 📋 Checklist

- [x] Rebased from the main (or release if patching) branch (before testing)
- [x] Ran `make test` before taking out of draft
- [x] Ran `make lint` before taking out of draft
- [x] Added/updated applicable tests
- [ ] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: cloud/auth/auth.go [hunk 1]
```diff
@@ -289,7 +290,7 @@ func CheckUserSession(c *config.Context, client astro.Client, coreClient astroco
 			break
 		}
 	}
-	err = c.SetOrganizationContext(activeOrg.Id, activeOrg.ShortName)
+	err = c.SetOrganizationContext(activeOrg.Id, activeOrg.ShortName, fmt.Sprintf("%s", *activeOrg.Product)) //nolint
```
Comment (jemishp) [inline cloud/auth/auth.go:293]: Do we need to check if `*activeOrg.Product` is nil? Or will it always have a value?

File: cloud/organization/organization.go [hunk 2]
```diff
@@ -115,8 +116,9 @@ func getOrganizationSelection(out io.Writer, coreClient astrocore.CoreClient) (*
 
 func SwitchWithContext(domain string, targetOrg *astrocore.Organization, astroClient astro.Client, coreClient astrocore.CoreClient, out io.Writer) error {
 	c, _ := context.GetCurrentContext()
+
 	// reset org context
-	_ = c.SetOrganizationContext(targetOrg.Id, targetOrg.ShortName)
+	_ = c.SetOrganizationContext(targetOrg.Id, targetOrg.ShortName, fmt.Sprintf("%s", *targetOrg.Product)) //nolint
```
Comment (jemishp) [inline cloud/organization/organization.go:121]: Same comment as above, do we need a nil check on `targetOrg.Product` before writing it to context?

**Review by jemishp — APPROVED** [review]
Left a couple of questions. But other than that looks good.

File: cloud/organization/organization.go (hunk 2 above)
Comment (kushalmalani) [reply cloud/organization/organization.go]: Added a check

**Pull request merged**